From 995a257b2c1716d99fd2ca5979282388f49ab606 Mon Sep 17 00:00:00 2001
From: o9000 <mrovi9000@gmail.com>
Date: Sun, 21 Feb 2016 19:58:30 +0100
Subject: [PATCH] Use standard macro to identify Linux (fixes build on powerpc)

---
 src/battery/linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/battery/linux.c b/src/battery/linux.c
index f36096b..54af607 100644
--- a/src/battery/linux.c
+++ b/src/battery/linux.c
@@ -17,7 +17,7 @@
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 **************************************************************************/
 
-#ifdef __linux
+#ifdef __linux__
 
 #include <stdlib.h>
 
--
libgit2 0.23.0

